### PR TITLE
Experiments: Update Card styles

### PR DESCRIPTION
**Pull request by mikewheaton**
#### Pull request checklist

- [x] Addresses an existing issue: N/A
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Adds tokens to Card to control padding, shadow, and width. Cards can take an onClick function, which also applies a broader shadow on hover. Compact cards use a horizontal stack.

This is a **work in progress** that contains a commented-out test that Mak is looking into.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8019)

### CODE REVIEW

File: packages/experiments/src/components/Card/CardItem/CardItem.types.ts [hunk 1]
```diff
@@ -18,6 +18,21 @@ export interface ICardItemProps extends ICardItemSlots, IStyleableComponentProps
...
+   */
+  grow?: boolean | number | 'inherit' | 'initial' | 'unset';
+
+  /**
+   * Defines at what ratio should the StackItem shrink to fit the available space.
+   */
+  shrink?: boolean | number | 'inherit' | 'initial' | 'unset';
+
+  /**
+   * Defines how to align the StackItem along the x-axis (for vertical Stacks) or the y-axis (for horizontal Stacks).
```
Comment (Vitalius1) [inline packages/experiments/src/components/Card/CardItem/CardItem.types.ts]: I am not following... what `StackItem` has to do with the `CardItemProps`

File: packages/experiments/src/components/Card/CardItem/CardItem.types.ts [hunk 2]
```diff
@@ -18,6 +18,21 @@ export interface ICardItemProps extends ICardItemSlots, IStyleableComponentProps
...
+   */
+  align?: 'auto' | 'stretch' | 'baseline' | 'flex-start' | 'center' | 'flex-end';
```
Comment (Vitalius1) [inline packages/experiments/src/components/Card/CardItem/CardItem.types.ts]: It almost feels we are reimplementing Stack (flexbox) props. 🤔

File: packages/experiments/src/components/Card/CardItem/CardItem.test.tsx [hunk 3]
```diff
@@ -54,26 +54,33 @@ describe('Card Item', () => {
     expect(tree).toMatchSnapshot();
   });
 
-  it('has the correct margin values when the disableChildPadding prop is set to true', () => {
-    const cardStylesFunc = CardStyles as IStylesFunction<ICardProps, ICardTokens, ICardStyles>;
-    const cardItemStylesFunc = CardItemStyles as IStylesFunction<ICardItemProps, ICardItemTokens, ICardItemStyles>;
+  // @todo: Get this test working.
```
Comment (Vitalius1) [inline packages/experiments/src/components/Card/CardItem/CardItem.test.tsx]: A lot of commenting in this file 😞

File: packages/experiments/src/components/Card/CardItem/CardItem.styles.ts [hunk 4]
```diff
@@ -21,7 +21,18 @@ export const CardItemStyles: ICardItemComponent['styles'] = (props, theme): ICar
       disableChildPadding && {
         marginLeft: -12,
         marginRight: -13
+      },
+      grow && { flexGrow: grow === true ? 1 : grow },
+      !grow &&
+        !shrink && {
+          flexShrink: 0
```
Comment (Vitalius1) [inline packages/experiments/src/components/Card/CardItem/CardItem.styles.ts]: Not knowing the internals and just curious... if `Stack` is used why we have all this flexbox specific styles here all over the place.

File: packages/experiments/src/components/Card/Card.styles.ts [hunk 5]
```diff
@@ -1,12 +1,33 @@
...
+};
+
+const clickableTokens: ICardComponent['tokens'] = {
+  boxShadowHovered: Depths.depth64
+};
+
+export const CardTokens: ICardComponent['tokens'] = (props, theme): ICardTokenReturnType => [
+  baseTokens,
+  props.compact && compactTokens,
+  props.onClick && clickableTokens
```
Comment (Vitalius1) [inline packages/experiments/src/components/Card/Card.styles.ts]: So if it's not clickable we don't have a hover state change of the box-shadow?

File: common/changes/@uifabric/experiments/miwhea-card-styles_2019-02-15-23-35.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@uifabric/experiments",
+      "comment": "Update Card styles and add tokens",
+      "type": "patch"
```
Comment (Vitalius1) [inline common/changes/@uifabric/experiments/miwhea-card-styles_2019-02-15-23-35.json]: Should be minor as you are adding new props.

File: packages/experiments/src/components/Card/CardItem/CardItem.types.ts (hunk 1 above)
Comment (khmakoto) [reply packages/experiments/src/components/Card/CardItem/CardItem.types.ts]: This is pretty much a WIP that we started today and we have to clean up moving forward. That said, `CardItem` shares many props with `StackItem`, to the point where we are thinking about extending `CardItem` from `StackItem`, though further discussion is needed.

File: packages/experiments/src/components/Card/CardItem/CardItem.types.ts (hunk 2 above)
Comment (khmakoto) [reply packages/experiments/src/components/Card/CardItem/CardItem.types.ts]: See above comment.

**Comment by khmakoto** [comment]
Adding `Do Not Merge` tag as this is pretty much a WIP.

File: packages/experiments/src/components/Card/Card.styles.ts (hunk 5 above)
Comment (khmakoto) [reply packages/experiments/src/components/Card/Card.styles.ts]: We have to follow up with design for that, but at the moment there is none 😄 .

File: packages/experiments/src/components/Card/CardItem/CardItem.styles.ts (hunk 4 above)
Comment (khmakoto) [reply packages/experiments/src/components/Card/CardItem/CardItem.styles.ts]: Again, most of this was just copy-pasting things from `Stack.Item`, will probably extend or leverage the `Stack.Item` component as we move forward with this PR.

File: packages/experiments/src/components/Card/CardItem/CardItem.test.tsx (hunk 3 above)
Comment (khmakoto) [reply packages/experiments/src/components/Card/CardItem/CardItem.test.tsx]: Yes, we realized this test has some troubles that we have to fix before merging this PR.

Comment (KevinTCoughlin) [reply packages/experiments/src/components/Card/CardItem/CardItem.test.tsx]: If you prefer, you can disable a test with `xit` and similarly `xdescribe`.

Comment (mikewheaton) [reply packages/experiments/src/components/Card/CardItem/CardItem.test.tsx]: I'll leave it for now as @khmakoto is going to be taking a closer look at this test.

File: common/changes/@uifabric/experiments/miwhea-card-styles_2019-02-15-23-35.json (hunk 6 above)
Comment (mikewheaton) [reply common/changes/@uifabric/experiments/miwhea-card-styles_2019-02-15-23-35.json]: Good catch, updated!

**Comment by mikewheaton** [comment]
> Adding `Do Not Merge` tag as this is pretty much a WIP at the moment.

Maybe we should be working off a shared branch at this point, rather than making PRs?

**Comment by mikewheaton** [comment]
@khmakoto, I'm going to close this PR and will push these changes to a new branch that we can work off of. Might as well wait and do a PR when it's not a work in progress.

**Pull request closed without merging**